### PR TITLE
WIP: RFC FS-1088 - if! (if-bang)

### PR DESCRIPTION
**Pull request by 3Rafal**
https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1088-if-bang.md

### CODE REVIEW

**Comment by vzarytovskii** [comment]
I've added a link to the RFC in the PR body

File: src/fsharp/SyntaxTree.fsi [hunk 1]
```diff
@@ -808,6 +808,17 @@ type SynExpr =
...
+    /// F# syntax: if! expr then expr
+    /// F# syntax: if! expr then expr else expr
+    | IfBangThenElse of
+        ifExpr: SynExpr *
+        thenExpr: SynExpr *
+        elseExpr: SynExpr option *
+        spIfToThen: DebugPointAtBinding *
+        isFromErrorRecovery: bool *
+        range: range *
+        trivia: SynExprIfThenElseTrivia
```
Comment (nojaf) [inline src/fsharp/SyntaxTree.fsi:820]: Please extend the `SynExprIfThenElseTrivia` to include `elif!` keyword where we today have `IsElif`

**Comment by edgarfgp** [comment]
@T-Gro Now that @kerams is working on the ```while!``` . Would be awesome to push for this also. having ``` while! and if!``` would be marvelous  :)

**Comment by KevinRansom** [comment]
@3Rafal ,

Thanks for this contribution, it looks as if it's been over a year since you have had time to work on it so I am closing it.  Please feel free to reopen this work if you get the time to work on it some more.

Thanks

Kevin

**Pull request closed without merging**